elastic/kibana
# [Ingest Pipelines] Add attachment processor

Fixes: https://github.com/elastic/kibana/issues/154395

## Summary

Adds support for the [attachment processor](https://www.elastic.co/guide/en/elasticsearch/reference/master/attachment.html).

**How to test**
1. Start elasticsearch with `yarn es snapshot --license=trial` and kibana with `yarn start`
2. Navigate to `Stack Management` -> `Ingest Pipelines` and create a new pipeline
3. Add new Attachment processor that has:
    * field: `data`
4. Verify that:
    * The request is correctly generated
    * Testing the pipeline with the following document should produce a valid attachment output
 ```
[
  {
    "found": true,
    "_index": "my-index-000001",
    "_id": "my_id",
    "_version": 1,
    "_seq_no": 22,
    "_primary_term": 1,
    "_source": {
      "data": "e1xydGYxXGFuc2kNCkxvcmVtIGlwc3VtIGRvbG9yIHNpdCBhbWV0DQpccGFyIH0="
    }
  }
]
 ```
Should output:
```
{
  "docs": [
    {
      "doc": {
        "_index": "my-index-000001",
        "_id": "my_id",
        "_version": "1",
        "_source": {
          "data": "e1xydGYxXGFuc2kNCkxvcmVtIGlwc3VtIGRvbG9yIHNpdCBhbWV0DQpccGFyIH0=",
          "attachment": {
            "content_type": "application/rtf",
            "language": "ro",
            "content": "Lorem ipsum dolor sit amet",
            "content_length": 28
          }
        },
        "_ingest": {
          "timestamp": "2023-04-19T07:18:56.021024Z"
        }
      }
    }
  ]
}
```

<details>
<summary>Screenshots</summary>

![Screenshot 2023-04-19 at 09 19 18](https://user-images.githubusercontent.com/1191206/232998374-fcc11c4c-3b2c-43bd-b089-5a710e232e12.png)
![Screenshot 2023-04-19 at 09 19 13](https://user-images.githubusercontent.com/1191206/232998381-b7e0c8c3-c50e-4415-9e07-a2a91630f9c9.png)



</details>
